ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/LedgerApiServer.scala

```diff
@@ -52,136 +65,99 @@ object LedgerApiServer {
       poolName = s"ledger-api-server-rs-grpc-bridge-${UUID.randomUUID}",
       actorCount = Runtime.getRuntime.availableProcessors() * 8
     )(mat.system)
+    closeables.push(serverEsf)
+
+    val workerEventLoopGroup = createEventLoopGroup(
+      mat.system.name + "-nio-worker",
+      parallelism = Runtime.getRuntime.availableProcessors)
+    val bossEventLoopGroup = createEventLoopGroup(mat.system.name + "-nio-boss", parallelism = 1)
+    closeables.push(() => {
+      // `shutdownGracefully` has a "quiet period" which specifies a time window in which
+      // _no requests_ must be witnessed before shutdown is _initiated_. Here we want to
+      // start immediately, so no quiet period -- by default it's 2 seconds.
+      // Moreover, there's also a "timeout" parameter
+      // which caps the time to wait for the quiet period to be fullfilled. Since we have
+      // no quiet period, this can also be 0.
+      // See <https://netty.io/4.1/api/io/netty/util/concurrent/EventExecutorGroup.html#shutdownGracefully-long-long-java.util.concurrent.TimeUnit->.
+      // The 10 seconds to wait is sort of arbitrary, it's long enough to be noticeable though.
+      Seq(
+        workerEventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS),
+        bossEventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS),
+      ).foreach(_.await(10, SECONDS))
+    })
 
     createApiServices(mat, serverEsf).map { apiServices =>
-      new ApiServer {
-        private val impl = new LedgerApiServer(
-          apiServices,
-          desiredPort,
-          maxInboundMessageSize,
-          address,
-          loggerFactory,
-          sslContext,
-          interceptors,
-          metrics
-        )
-
-        /** returns the api port the server is listening on */
-        override def port: Int = impl.port
-
-        /** returns when all services have been closed during the shutdown */
-        override def servicesClosed(): Future[Unit] = impl.servicesClosed()
-
-        override def close(): Unit = {
-          impl.close()
-          serverEsf.close()
+      val builder = address.fold(NettyServerBuilder.forPort(desiredPort))(address =>
+        NettyServerBuilder.forAddress(new InetSocketAddress(address, desiredPort)))
+      sslContext
+        .fold {
+          logger.info("Starting plainText server")
```

Comment:
Would it make sense to explicitly call the method to set the channel to use an unsecured channel here? Just in case the default changes or this piece of code gets moved or reused without having the builder under control.

Reply:
If you can tell me what that method is, sure. I can't find it. 😉

Reply:
Uh, looks like it's not defined over this builder. From the code it looks like you should call `sslContext` with a `null` argument, but I would ask to double check whether that's the case.

Reply:
I did exactly that, and moved the log line to simplify things.